src/test/java/de/tum/in/www1/artemis/util/UserTestService.java

```diff
@@ -927,4 +932,54 @@ public void testUserWithExternalAndInternalStatus() throws Exception {
             assertThat(result).isEqualTo(Collections.emptyList());
         }
     }
+
+    /**
+     * Test for the user search with the following parameters:
+     * @Param registrationNumbers
+     * @throws Exception if the user is not the same as the expected user
+     */
+    public void testUserWithRegistrationNumber() throws Exception {
+        final var params = createParamsForPagingRequest("USER", "INTERNAL,EXTERNAL", "WITH_REG_NO", "", "");
+
+        List<User> result;
+        List<User> users;
+
+        database.addEmptyCourse();
+
+        int[][] numbers = { { 2, 0, 0, 0 }, { 0, 2, 0, 0 }, { 0, 0, 2, 0 }, { 0, 0, 0, 2 } };
+        for (int[] number : numbers) {
+            userRepository.deleteAll();
+            users = database.addUsers(number[0], number[1], number[2], number[3]).stream().peek(user -> user.setGroups(Collections.emptySet())).toList();
+            users.get(0).setRegistrationNumber("");
+            users.get(1).setRegistrationNumber("");
+            userRepository.saveAll(users);
+            result = request.getList("/api/users", HttpStatus.OK, User.class, params);
+            assertThat(result).isEqualTo(Collections.emptyList());
+        }
+    }
+
+    /**
+     * Test for the user search with the following parameters:
+     * @Param registrationNumbers
+     * @throws Exception if the user is not the same as the expected user
+     */
+    public void testUserWithoutRegistrationNumber() throws Exception {
```

Comment:
❓Question: Just to confirm. You are also testing the case here where the registration number is null right?

Reply:
It worked for both cases